### PR TITLE
[Fix] Fix inference api and support setting palette to SegLocalVisualizer

### PR DESCRIPTION
**Pull request by xiexinch**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMDet3D.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmseg/visualization/local_visualizer.py [hunk 1]
```diff
@@ -55,14 +58,61 @@ def __init__(self,
...
+        elif isinstance(classes, str):
+            try:
+                classes = get_classes(classes)
+            except ValueError:
+                print_log(
+                    f'Unrecognized dataset: {classes}, '
+                    f'expect classes in {dataset_aliases}',
+                    logger='current',
+                    level=logging.ERROR)
+                exit(-1)
```
Comment (MeowZheng) [inline mmseg/visualization/local_visualizer.py:90]: Does `classes` mean dataset name here?

File: mmseg/visualization/local_visualizer.py [hunk 2]
```diff
@@ -55,14 +58,61 @@ def __init__(self,
...
+        palette = self._init_palette(palette)
+        if classes is not None and palette is not None:
+            assert len(classes) == len(
+                palette), 'The length of classes should be equal to palette'
+        self.dataset_meta: dict = {classes: classes, palette: palette}
+
+    def _init_classes(
+            self, classes: Optional[Union[str, List]]) -> Union[List, None]:
+        if classes is None:
+            classes = None
```
Comment (MeowZheng) [inline mmseg/visualization/local_visualizer.py:80]: it's a little redundant

**Review by MeowZheng — COMMENTED** [review]
I think here we need to provide a proper way for users to set the classes and palette of the visualizer, but input arguments `classes` and `palette` are a little unclear now.

File: mmseg/visualization/local_visualizer.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
-# Copyright (c) OpenMMLab. All rights reserved.
-from typing import Dict, List, Optional, Tuple
+# Copyright (c) OpenMMLab. All rights reserved.g
```
Comment (MengzhangLI) [inline mmseg/visualization/local_visualizer.py:1]:
```suggestion
# Copyright (c) OpenMMLab. All rights reserved.
```
It might be a typo.

**Review by MeowZheng — APPROVED** [review]

**Pull request merged**